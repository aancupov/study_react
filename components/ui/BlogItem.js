import React from 'react';
import DOM from 'react-dom-factories'
import PropTypes from 'prop-types'
import Image from './Image'
import Message from './Message'
import MetaInfo from './MetaInfo'

const BlogItem = ({ image, message, metainfo }) => (
  DOM.div(
    null
    , React.createElement(Image,    image)
    , React.createElement(Message,  message)
    , React.createElement(MetaInfo, metainfo)  
  )
);

BlogItem.propTypes = {
  message:    PropTypes.shape( { text: PropTypes.string }),
  image:  PropTypes.shape(
    {
      src:    PropTypes.string,
      width:  PropTypes.string,
      height: PropTypes.string,
      alt:    PropTypes.string     
    }
  )
}

export default BlogItem;
