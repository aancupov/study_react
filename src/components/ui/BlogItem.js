import React from 'react';

import PropTypes from 'prop-types';

import Image from './Image';
import Message from './Message';
import MetaInfo from './MetaInfo';
import LikesContainer from 'containers/LikesContainer';

import 'css/app.css';

import { Link } from 'react-router-dom';

import { postPath } from 'helpers/routes';

const BlogItem = ({ item }) => (
  <div className='blog-item'>
    <Image {...item.image } />
    <Message><Link to={postPath(item.id)}>{ item.message }</Link></Message>
    <MetaInfo {... item.metainfo } />
    <LikesContainer id={item.id}/>
  </div>
);

BlogItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number, 
    message: Message.PropTypes, 
    image: Image.PropTypes
  })
};

export default BlogItem;
