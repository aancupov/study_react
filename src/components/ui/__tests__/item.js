import React from 'react';

import ReactDOM from 'react-dom';

import  { Router } from  'react-router-dom';

import createBrowserHistory from 'history/createBrowserHistory';

import { Provider } from 'react-redux';

import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Link from 'components/elements/Link';

import BlogItem from '../BlogItem';

describe('BlogItem', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store = {
      subscribe: () => {},
      dispatch: () => {},
      getState: () => ({ 
        likes: {
          error: false,
          isFetching: true,
          resultLikes: [
            {id: 0, likes: 0},
            {id: 1, likes: 0},
            {id: 2, likes: 0},
            {id: 3, likes: 0}
          ]
        },
        post: {
          isFetching: true, 
          isUpdating: false, 
          isAdding: false, 
          error: false, 
          entry: {
            id: 3,
            image: {
              src: '/images/world.png', 
              width: '100px', 
              height: '100px', 
              alt: 'Bin'
            },
            message: 'У-у-уу',
            metainfo: {
              author: 'Petr', 
              createdAt: '2017-01-01', 
              updatedAt: '2017-01-02'
            } 
          }
        },
        posts: {
          isFetching: false, 
          error: false, 
          entries: [], 
          pagination: 0
        }        
      })
    };
    ReactDOM.render(
      <Provider store={store}>
        <Router history={createBrowserHistory()}>
          <BlogItem />
        </Router>
      </Provider>  
      , div);
  });

  describe('render', () => {
    it('should render the title', () => {
      const itemProps = { message: 'О-о-оо', id: 3 };
      const item = shallow(<BlogItem item={itemProps} />);
      const header = <Link to="/posts/3">О-о-оо</Link>;
      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const itemProps = {
        title: 'О-о-оо',
        id: 3,
        image: {
          alt: 'Hello, World',
          src: 'images/earth.png'
        }
      };

      const item = shallow(<BlogItem item={itemProps} />);

      expect(item).toMatchSnapshot();
    });
  }); 
});