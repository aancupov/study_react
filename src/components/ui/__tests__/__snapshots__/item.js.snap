// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem render should render usual item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlogItem
    item={
      Object {
        "id": 3,
        "image": Object {
          "alt": "Hello, World",
          "src": "images/earth.png",
        },
        "title": "О-о-оо",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt="Hello, World"
          height="20px"
          src="images/earth.png"
          width="50px"
        />,
        <Message>
          <Link
            replace={false}
            to="/posts/3"
          />
        </Message>,
        <MetaInfo
          author="admin"
          createdAt="2000-01-01"
          updatedAt="2000-01-01"
        />,
        <Connect(Likes)
          id={3}
        />,
      ],
      "className": "blog-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "Hello, World",
          "height": "20px",
          "src": "images/earth.png",
          "width": "50px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Link
            replace={false}
            to="/posts/3"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "replace": false,
            "to": "/posts/3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": "admin",
          "createdAt": "2000-01-01",
          "updatedAt": "2000-01-01",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt="Hello, World"
            height="20px"
            src="images/earth.png"
            width="50px"
          />,
          <Message>
            <Link
              replace={false}
              to="/posts/3"
            />
          </Message>,
          <MetaInfo
            author="admin"
            createdAt="2000-01-01"
            updatedAt="2000-01-01"
          />,
          <Connect(Likes)
            id={3}
          />,
        ],
        "className": "blog-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Hello, World",
            "height": "20px",
            "src": "images/earth.png",
            "width": "50px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Link
              replace={false}
              to="/posts/3"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "replace": false,
              "to": "/posts/3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": "admin",
            "createdAt": "2000-01-01",
            "updatedAt": "2000-01-01",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
